src/nccl_ofi_net.c

```diff
@@ -1237,7 +1237,10 @@ static ncclResult_t ofi_getProperties(int dev, ncclNetProperties_t *props)
 		goto exit;
 	}
 
-	dev_props.name = strdup(nic_info->device_attr->name);
+	/* name is NULL if device is a part of multirail config */
+	if (nic_info->device_attr->name) {
```

Comment:
I understand this this is used by NCCL for logging purposes. Is it possible to derive a name for multi-rail devices?

Reply:
Seconding this suggestion.  Maybe something like "providername unnamed device".  Also, are we guaranteed to have only one virtual device?  If not, we would probably want to include an index in the name also.

Reply:
Please take a look at new version